File: 2024-2/0682/unidad06.d_Web_04/SoftPub/SoftPubGrupoBussiness/GrupoBo.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using SoftPubGrupoControler.DAO;
using SoftPubGrupoControler.DAOImpl;
using SoftPubGrupoModel;

namespace SoftPubGrupoBussiness
{
    public class GrupoBo
    {
        private GrupoDAO grupoDAO;

        public GrupoBo()
        {
            this.grupoDAO = new GrupoDAOImpl();
        }

        public int? insertar(String nombre, String sigla)
        {
            Grupo grupo = new Grupo(null, nombre, sigla);
            return this.grupoDAO.insertar(grupo);
        }

        public BindingList<Object> listarTodos()
        {
            return this.grupoDAO.listarTodos();
        }

        public BindingList<Object> listarGruposConProduccion()
        {
            return this.grupoDAO.listarGruposConProduccion();
        }

        public BindingList<Object> buscarGrupos(string nombre)
        {
            return this.grupoDAO.buscarGrupos(nombre);
        }

        public Grupo obtenerPorId(int idGrupo)
        {
            return this.grupoDAO.obtenerPorId(idGrupo);
        }
    }
}

File: 2024-2/Proyecto DAO SoftPub Java/SoftPubGrupoBO/src/pe/edu/pucp/softpub/grupo/bo/GrupoBO.java
package pe.edu.pucp.softpub.grupo.bo;

import java.util.ArrayList;
import pe.edu.pucp.softpub.grupo.dao.GrupoDAO;
import pe.edu.pucp.softpub.grupo.daoImpl.GrupoDAOImpl;
import pe.edu.pucp.softpub.grupo.model.Grupo;

public class GrupoBO {

    private final GrupoDAO grupoDAO;
    
    public GrupoBO(){
        this.grupoDAO = new GrupoDAOImpl();
    }
    
    public Integer insertar(String nombre, String sigla){
        Grupo grupo = new Grupo(null, nombre, sigla);
        return this.grupoDAO.insertar(grupo);
    }
    
    public Integer modificar(Integer idGrupo, String nombre, String sigla){
        Grupo grupo = new Grupo(idGrupo, nombre, sigla);
        return this.grupoDAO.modificar(grupo);
    }
    
    public Integer eliminar(Integer idGrupo){
        Grupo grupo = new Grupo();
        grupo.setIdGrupo(idGrupo);
        return this.grupoDAO.eliminar(grupo);
    }
    
    public ArrayList<Grupo> listarTodos(){
        return this.grupoDAO.listarTodos();
    }
    
    public Grupo obtenerPorId(Integer idGrupo){
        return this.grupoDAO.obtenerPorId(idGrupo);
    }
}
